gnucash/gnome/dialog-invoice.c

```diff
@@ -501,6 +503,54 @@ gnc_invoice_window_help_cb (GtkWidget *widget, gpointer data)
     gnc_gnome_help(HF_HELP, HL_USAGE_INVOICE);
 }
 
+static gchar *
+gnc_invoice_window_get_state_group (InvoiceWindow *iw)
+{
+    switch (gncOwnerGetType (&iw->owner))
+    {
+        case GNC_OWNER_VENDOR:
+            return g_strdup ("Bill Default State Group");
```

Comment:
So to complete the whole story: these g_strdups should go if we want to make the state_group string const throughout the code.
What I *am* not sure about is whether these strings should be made static top-level strings or can be returned directly from the function here.

Reply:
They have to be declared `static` somewhere to be passable, otherwise the compiler will just create a temporary and return a ptr to it, but the ptr won't be valid because the temporary will have gone out of scope. It's OK to do the static decl in the function so that the strings don't pollute the compilation-unit's name space:
```C
#include <stdio.h>
#include <stddef.h>

const char*
string_func(int which)
{
     static const char* string1 = "I'm the first string!";
     static const char* string2 = "I'm the second string!";
     static const char* string3 = "I'm the last string!";

     switch (which)
     {
     case 0:
          return string1;
     case 1:
          return string2;
     case 2:
          return string3;
     default:
          return NULL;
     }
}

int
main(int argc, char** argv)
{
     for (int i = 0; i < 3; ++i)
          printf("%s\n", string_func(i));
     return 0;
}
```

Reply:
I just tried without the static decl and it still works:
```C
#include <stdio.h>
#include <stddef.h>

const char*
string_func(int which)
{
     switch (which)
     {
     case 0:
          return "I'm the first string!";
     case 1:
          return "I'm the second string!";
     case 2:
          return  "I'm the last string!";
     default:
          return NULL;
     }
}

void
print_string(const char* string)
{
     printf("%s\n", string);
}

int
main(int argc, char** argv)
{
     for (int i = 0; i < 3; ++i)
     {
          const char *string = string_func(i);
          print_string(string);
     }
     return 0;
}
```
So it seems that the compiler (clang) makes a static const char* from the literal without being told to.